backend/curriculum_tracking/tests/test_activity_log_entry_creators.py

```diff
@@ -23,6 +41,32 @@ def test_that_only_one_entry_created_for_one_review(self):
         creators.log_project_competence_review_done(review2)
         self.assertEqual(LogEntry.objects.count(), 2)
 
+    def test_two_entries_created_for_two_reviews_with_second_done_within_debounce_period_by_another_user(self):
+        review1 = RecruitProjectReviewFactory()
+        review2 = RecruitProjectReviewFactory(
+            status=review1.status,
+            timestamp=review1.timestamp,
+            comments=review1.comments,
+            recruit_project=review1.recruit_project,
+            reviewer_user=UserFactory()
+        )
+        creators.log_project_competence_review_done(review1)
+        creators.log_project_competence_review_done(review2)
+        self.assertEqual(LogEntry.objects.count(), 2)
+
+    def test_logging_the_same_entry_twice_only_produces_one_log_entry(self):
+        review1 = RecruitProjectReviewFactory()
+        creators.log_project_competence_review_done(review1)
+        creators.log_project_competence_review_done(review1)
+        self.assertEqual(LogEntry.objects.count(), 1)
+
+    def test_log_project_creator_is_invoked_for_a_project_review_else_attribute_error_is_raised(self):
+        review1 = RecruitProjectReviewFactory()
+        with self.assertRaises(AttributeError) as exception:
+            creators.log_topic_competence_review_done(review1)
+        self.assertTrue(exception, AttributeError)
+        self.assertEqual(LogEntry.objects.count(), 0)
```

Comment:
Not sure I get what's going on with this test. We're intentionally logging a project as a topic, and checking that an exception was raised. Is this actually related to the creator thats being tested here or is it more general?
This class is testing a specific creator, does this mean for every single creator there needs to be a test like this?

Reply:
If we have a topic review then accidentally calling "log_project_competence_review_done" should not work cause it deals with projects and not topics.  If it does work then we have a big problem since the data we are passing to the function has somehow become corrupt.  So this is kinda like a sanity check.